tests/test_geometry.py

```diff
@@ -0,0 +1,534 @@
+
+
+def test_baran_isoindole():
+    mol = Chem.AddHs(Chem.MolFromSmiles("c1cccc2c1c[nH]c2"))
+    atom_types = geometry.classify_geometry(mol)
+    expected_types = [
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+    ]
+
+    assert atom_types == expected_types
+
+
+def test_baran_carbazole():
+    mol = Chem.AddHs(Chem.MolFromSmiles("c1ccc2c(c1)c3ccccc3[nH]2"))
+    atom_types = geometry.classify_geometry(mol)
+    expected_types = [
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+    ]
+
+    assert atom_types == expected_types
+
+
+def test_baran_furan():
+    mol = Chem.AddHs(Chem.MolFromSmiles("c1ccco1"))
+    atom_types = geometry.classify_geometry(mol)
+    expected_types = [
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G2_KINK,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+    ]
+
+    assert atom_types == expected_types
+
+
+def test_baran_thiophene():
+    mol = Chem.AddHs(Chem.MolFromSmiles("c1cccs1"))
+    atom_types = geometry.classify_geometry(mol)
+    expected_types = [
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G2_KINK,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+    ]
+
+    assert atom_types == expected_types
+
+
+def test_baran_2h_pyran():
+    mol = Chem.AddHs(Chem.MolFromSmiles("C1C=CC=CO1"))
+    atom_types = geometry.classify_geometry(mol)
+    expected_types = [
+        LG.G4_TETRAHEDRAL,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G2_KINK,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+        LG.G1_TERMINAL,
+    ]
+
+    assert atom_types == expected_types
+
+
+def test_baran_pyridine():
+    mol = Chem.AddHs(Chem.MolFromSmiles("c1ccccn1"))
+    atom_types = geometry.classify_geometry(mol)
+    expected_types = [
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G3_PLANAR,
+        LG.G2_KINK,
```

Comment:
Why is it Kink and not Planar for the N in pyridine?

Reply:
the N in Pyridine has two neighbors: C=N-C, which means it can only be assigned to "Kink" or "Linear", and this certainly isn't linear.